wp-includes/class-wp-https-ui.php

```diff
@@ -79,6 +79,10 @@ public function __construct( $wp_https_detection ) {
 	 * Initializes the object.
 	 */
 	public function init() {
+		if ( apply_filters( WP_HTTPS_Detection::FILTER_DISABLE, false ) ) {
+			return;
+		}
+
```

Comment:
Is this necessary here? The `WP_HTTPS_Detection::init()` method is what instantiates this class, so the filter there should suffice.

Reply:
Not strictly, I was being cautious/paranoid. If the `WP_HTTPS_Detection` class is disabled but something instantiated the `WP_HTTPS_UI` class, it would add UI that doesn't do anything (since the redirect relies on data from `WP_HTTPS_Detection`) and I wanted to avoid any potential confusion.